data_types.md

```diff
@@ -1 +1,261 @@
-## Eldor, yangi branch yaratib, shu matnni o'chirib o'rniga maqolangizni yozing.
+# Ma'lumotlar turlari (Data types)
+<img src="https://i.ibb.co/YkPGqK9/assja.jpg" />
+
+
+
+  Dasturlashda data tiplari muhim tushuncha hisoblanadi. O'zgaruvchilar ustida ishlashni ta'minlash uchun uning tipi haqida ma'lumotga ega bo'lishimiz muhimdir. Data tiplarisiz kompyuter buni xavfsiz hal qila olmaydi.
+
+<img src="https://i.ibb.co/54h84xX/image.jpg" />
+
+
+  Ko'rib turganingizdek data tiplari 2 turga bo'linadi : 
+  
+  
+  1. **Primitive**  
+
+
+  2. **Non-primitive**
+  
+  
+  
+  **Primitiv** tiplarga quyidagilar kiradi :
+  `Number, String, Boolean, Null, Undefined, Symbol va BigInt`;
+  
+  
+  
+  **Non-primitive** tiplarga quyidagilar kiradi:
+  `Object`;
+  
+  **Primitiv**
+  tiplarning barchasi faqatgina yagona qiymatni qabul qiladi va ularga ikkinchi
+  qiymatni qo’sha olmaymiz. 
+  
+  
+  Keling bularni birin-ketinlik bilan ko'rib chiqamiz!
+
+### Number
+
+   Number -  64 bitli ikkilik formatdagi IEEE 754 ning qiymati (- (2 ^ 53 - 1) va 2 ^ 53 - 1 orasidagi sonlar).
+       Sonlar turi quyidagi 3 ta ko'rinishga ega bo'ladi: `+ Infinity, -Infinity va NaN ("Raqam emas").`
```

Comment:
NaN ("Not a Number" - "Raqam emas")

Manimcha NaNni inglizchasini ham aytib ketgan yaxshi

Reply:
xop to'g'irlab qo'yaman